### DISCUSSION TITLE
Error handling when creating Telegraf instance with incorrect token

### DISCUSSION THREAD

**Discussion by hodlerhacks** [telegraf/telegraf#1474] [Q&A]
My application allows users to enter a token for their own Telegraf bot. My problem is that when creating a new Telegraf instance with an incorrect token, I cannot catch the error (Error: 401: Unauthorized) which is thrown. Putting a normal try-catch around the new Telegraf(<TOKEN>) doesn't work. Is there any other way to do this? Thanks!

**Comment by jsstrn** [comment]
Have you tried accessing a property to trigger an error?

```js
try {
  const tg = new Telegraf('invalid-bot-token');
  tg.botInfo // this should trigger an error if its invalid
} catch(error) {
  // handle error here
}
```

Or you could explicitly throw an error if the property is not accessible: 

```js
if (!tg.botInfo) {
  throw new Error('Gasp!')
}
```

**Reply by hodlerhacks** [reply]
Thanks, great suggestion!
